cppcache/integration/test/PositionKey.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "PositionKey.hpp"
+
+#include <wchar.h>
+
+#include <cwchar>
+
+#include <geode/DataInput.hpp>
+#include <geode/DataOutput.hpp>
+
+namespace DataSerializableTest {
+
+void PositionKey::toData(DataOutput& output) const {
+  output.writeInt(m_positionId);
+}
+
+void PositionKey::fromData(apache::geode::client::DataInput& input) {
+  m_positionId = input.readInt64();
+}
+
+bool PositionKey::operator==(const CacheableKey& other) const {
+  return m_positionId == ((PositionKey&)other).getPositionId();
+}
+
+int PositionKey::hashcode() const {
+  int hash = 11;
+  hash = 31 * hash + (int)m_positionId;
```

Comment:
This hash function does not match the Java version of this type. This should be fixed or partitioning will be off. Also very odd both just down cast `positionId`. Regardless you should use `static_cast<int32_t>()` not C-style casts.

Reply:
Now they match, and switched to static_cast.